File: leetcode/0001. 两数之和/main.go
package main

import "fmt"

func main() {
	fmt.Println(twoSum([]int{1, 2, 3, 5}, 5))
}

func twoSum(nums []int, target int) []int {
	for i, m := range nums {
		for j, n := range nums[i+1:] {
			if m+n == target {
				return []int{i, j + i + 1}
			}
		}
	}
	return nil
}

File: leetcode/0001. 两数之和/main.ts
function twoSum1(nums: number[], target: number): number[] {
    let map = new Map<number, number>();
    for (let i = 0; i < nums.length; i++) {
        let diff: number = target - nums[i];
        if (map.has(diff)) {
            return [map.get(diff) as number, i];
        }
        map.set(nums[i], i);
    }
    return [];
}

(function main() {
    console.log(twoSum1([1, 2, 3], 3));
})()
